cpp/src/arrow/python/flight.cc

```diff
@@ -245,6 +245,19 @@ Status CreateFlightInfo(const std::shared_ptr<arrow::Schema>& schema,
   return Status::OK();
 }
 
+Status DeserializeBasicAuth(const std::string& buf,
+                            std::unique_ptr<arrow::flight::BasicAuth>* out) {
+    auto basic_auth = new arrow::flight::BasicAuth();
+    *out = std::unique_ptr<arrow::flight::BasicAuth>(basic_auth);
+    return arrow::flight::BasicAuth::Deserialize(buf, basic_auth);
+}
+
+Status SerializeBasicAuth(const arrow::flight::BasicAuth &basic_auth, std::unique_ptr<std::string> *out) {
```

Comment:
Do we need `unique_ptr<string>`? `string*` should suffice (and that's what the other serialization methods do).

Reply:
both have been fixed. Thanks @lihalite